### DISCUSSION TITLE
With exactly once handler seems to warn of sending ack even on success

### DISCUSSION THREAD

**Discussion by pantoss** [googleapis/google-cloud-cpp#10917] [Q&A] [answered]
I'm using tag 2.3.0 but it look like the behaviour will be the same on the tip of master.

From the code at https://github.com/googleapis/google-cloud-cpp/blob/5ef3ff728f734e015df83bd6aa8690b146576637/google/cloud/pubsub/internal/ack_handler_wrapper.cc#L27

Even if the ack is returns a status of OK you will get the warning like this:
'Error while trying to ack(), status=OK, message_id=6951860352329453'

Is this correct? Am I miss interpreting what status=OK means?

**Comment by coryan** [comment, marked as answer]
That looks like a bug, let me send a PR.

**Reply by coryan** [reply]
I sent #10920 to fix this.  Thanks for raising the issue.  I expect the PR will get merged later today, and will be part of the upcoming release in early 2023-03.
